ccm_web/server/src/api/api.controller.ts

```diff
@@ -90,6 +91,31 @@ export class APIController {
     return result
   }
 
+  @UseInterceptors(InvalidTokenInterceptor)
+  @ApiSecurity('CSRF-Token')
+  @Post('course/:id/sections/enroll')
+  async enrollUsersToSections (
+    @Param('id', ParseIntPipe) courseId: number, @Body() enrollmentsDto: SectionEnrollmentsDto, @UserDec() user: User
+  ): Promise<CanvasEnrollment[]> {
+    const enrollments = enrollmentsDto.enrollments
+    const getSectionsResult = await this.apiService.getCourseSections(user, courseId)
+    if (isAPIErrorData(getSectionsResult)) throw new HttpException(getSectionsResult, getSectionsResult.statusCode)
+    const validSectionIds = getSectionsResult.map(s => s.id)
+    const invalidSectionIds: number[] = []
+    enrollments.forEach(e => {
+      if (!validSectionIds.includes(e.sectionId) && !invalidSectionIds.includes(e.sectionId)) {
+        invalidSectionIds.push(e.sectionId)
+      }
+    })
+    if (invalidSectionIds.length > 0) {
+      const message = `Course with ID ${courseId} does not have sections with the following ID(s): ${invalidSectionIds.join(', ')}`
+      throw new BadRequestException(message)
+    }
```

Comment:
I was trying to see a scenario where the frontend validation missed identifying the course sections vs other course sections, if don't see that is possible. So to me this additional check if all these sections belong to course is redundant. The FE is loading latest section info appropriately.

Reply:
I guess I was just trying to be thorough, and protect the API endpoint from the same potential validation problem. I see your perspective though, the Canvas token will limit the user to editing their sections (across all their courses). I can remove it.

Reply:
Yes... Scoped token approach is turning out as good choice for most of the time

Reply:
Okay, @pushyamig, see here: https://github.com/tl-its-umich-edu/canvas-course-manager-next/pull/301/commits/5c714c6571ba715af719191b5d27581061debced